File: tests/tester.rb
#!/usr/bin/env ruby

$LOAD_PATH << '../ruby'

require 'rubygems'
require 'om'
require 'cgi'

consumer = [ARGV[0], ARGV[1]]
url = ARGV[2]
params = CGI.parse(ARGV[3])
params.each_pair{|k,v| params[k] = v[0]}
if ARGV[4] != '' and ARGV[5]
  token = [ARGV[4], ARGV[5]]
else
  token = nil
end
method = ARGV[6]
if ARGV[7] != ''
  realm = ARGV[7]
else
  realm = nil
end
timestamp = ARGV[8]
nonce = ARGV[9]
print om(consumer, url, params, token, method, realm, timestamp, nonce)

File: tests/tester.php
#!/usr/bin/php -q
<?
require('../php/om.php');

$consumer = array($argv[1], $argv[2]);
$url = $argv[3];
$params = array();
parse_str($argv[4], $params);
if ($argv[5] != '' && $argv[6] != '') {
  $token = array($argv[5], $argv[6]);
} else {
  $token = NULL;
}
$method = $argv[7];
if ($argv[8] != '') {
  $realm = $argv[8];
} else {
  $realm = NULL;
}
$timestamp = $argv[9];
$nonce = $argv[10];
print om($consumer, $url, $params, $token, $method, $realm, $timestamp, $nonce);
?>